Modules/posixmodule.c

```diff
@@ -13244,20 +13244,9 @@ os_get_terminal_size_impl(PyObject *module, int fd)
 
 #ifdef TERMSIZE_USE_CONIO
     {
-        DWORD nhandle;
         HANDLE handle;
         CONSOLE_SCREEN_BUFFER_INFO csbi;
-        switch (fd) {
-        case 0: nhandle = STD_INPUT_HANDLE;
-            break;
-        case 1: nhandle = STD_OUTPUT_HANDLE;
-            break;
-        case 2: nhandle = STD_ERROR_HANDLE;
-            break;
-        default:
-            return PyErr_Format(PyExc_ValueError, "bad file descriptor");
-        }
-        handle = GetStdHandle(nhandle);
+        handle = (HANDLE)_get_osfhandle(fd);
```

Comment:
By default `_get_osfhandle(fd)` will abort the process if `fd` isn't valid. That's why my example wrapped the call with `_Py_BEGIN_SUPPRESS_IPH` and `_Py_END_SUPPRESS_IPH`. Also, the error is based on `errno` in this case, which is why my example used `PyErr_SetFromErrno(PyExc_OSError)`. 

In Python 3.10+, the internal API has `_Py_get_osfhandle()` to hide these implementation details. For example:

```suggestion
        handle = _Py_get_osfhandle(fd);
        if (handle == INVALID_HANDLE_VALUE) {
            return NULL;
        }
```
Remove the following error handling code:

```
        if (handle == NULL)
            return PyErr_Format(PyExc_OSError, "handle cannot be retrieved");
        if (handle == INVALID_HANDLE_VALUE)
            return PyErr_SetFromWindowsErr(0);
```

Reply:
Thanks for the suggestion. 
However, using `_Py_get_osfhandle()` will still abort the process when python is built in debug mode.

Reply:
In a debug build, most CRT functions report assertion failures, warnings, and errors according to the configured report mode. The default report mode shows a dialog that prompts to abort, retry (break into a debugger to examine local variables and single step), or ignore a reported problem. Python's test suite uses `msvcrt.CrtSetReportMode()` and `msvcrt.CrtSetReportFile()` to configure the CRT to passively report problems by writing a message to `stderr`.

Reply:
The assert/warning/error report mode in debug builds is unrelated to the CRT's invalid parameter handler, which always gets called by CRT functions that check for valid parameter values. The default handler aborts the process using the [`__fastfail`](https://docs.microsoft.com/en-us/cpp/intrinsics/fastfail?view=msvc-170) intrinsic, which we usually do not want in Python. Thus we use the `_Py_BEGIN_SUPPRESS_IPH` macro to set an invalid parameter handler for the current thread that does nothing, followed by `_Py_END_SUPPRESS_IPH` to restore the previous handler. Typically the CRT function will fail with `errno` set to `EINVAL`, `EBADF`, etc, which is want we want in order to raise an exception that can be handled by Python code.